ydb/core/tx/schemeshard/schemeshard_info_types.cpp

```diff
@@ -1577,19 +1577,21 @@ void TAggregatedStats::UpdateShardStats(TShardIdx datashardIdx, const TPartition
     Aggregated.IndexSize += (newStats.IndexSize - oldStats.IndexSize);
     for (const auto& [poolKind, newStoragePoolStats] : newStats.StoragePoolsStats) {
         auto* aggregatedStoragePoolStats = Aggregated.StoragePoolsStats.FindPtr(poolKind);
+        const auto* oldStoragePoolStats = oldStats.StoragePoolsStats.FindPtr(poolKind);
         if (aggregatedStoragePoolStats) {
-            const auto* oldStoragePoolStats = oldStats.StoragePoolsStats.FindPtr(poolKind);
-
             // Missing old stats for a particular storage pool are interpreted as if this data
             // has just been written to the datashard and we need to increment the aggregate by the entire new stats' sizes.
             aggregatedStoragePoolStats->DataSize += newStoragePoolStats.DataSize - (oldStoragePoolStats ? oldStoragePoolStats->DataSize : 0u);
             aggregatedStoragePoolStats->IndexSize += newStoragePoolStats.IndexSize - (oldStoragePoolStats ? oldStoragePoolStats->IndexSize : 0u);
         } else {
+            Y_ABORT_UNLESS(!oldStoragePoolStats, "Old stats are present, but they haven't been aggregated.");
```

Comment:
I added this Y_ABORT_UNLESS for greater similarity with the for loop below (which I have added before).

I believe that the state of the TAggregatedStats in which:
1) the aggregated stats do not contain an entry for a particular storage pool kind
2) there is an old stats entry for this storage kind

cannot be correctly handled silently. The semantics of TEvPeriodicStats messages implies that the aggregated stats are incremented by `newStats - oldStats (or 0 if they are absent)`. If some old stats, that haven't been aggregated, are present, it signals an error in aggregation, which should be investigated.

I suppose it is better to discover these kinds of errors during testing deployment than to silently ignore and calculate an aggregate using some stale oldStats.

Reply:
But also we don't want to lose entire node (with schemeshard and all other tablets) to a cycle of the same aborts. `Y_ABORT_UNLESS` are too risky in that regard.